modules/core/src/utils/query-object.js

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2015 - 2017 Uber Technologies, Inc.
```

Comment:
Nit: filename is a bit generic. Idea: `query-picking-buffer`?

Reply:
Hmm, are there no tests for these functions? 

I would expect to see some change in test case file structure? Maybe create an placeholder test file that at least test that these function are correctly exported?

Reply:
there are tests, I didn't remove the old pick-layers.js file, so didn't work on the tests yet.